### PR TITLE
Add swerve drive

### PR DESCRIPTION
**Pull request by dracco1993**
~~Still not _quite_~~ ready, but might as well at least start looking at the `diff`.   👀

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
+    // TODO: Change for final robot
+    public static final double kXDistance = 0.762; // 30 inches
+    public static final double kYDistance = 0.762; // in meters!
+
+    public static final double kXCenterDistance = kXDistance / 2.0;
+    public static final double kYCenterDistance = kYDistance / 2.0;
+
+    // Drivetrain drive motor constants
+    public class Drive {
+      public static final int kFLDriveMotorId = 5;
```
Comment (snydecor000) [inline src/main/java/frc/robot/Constants.java]:
```suggestion
      public static final int kFLMotorId = 5;
```

Remove redundant naming when inside of a scope with that name

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -1,36 +1,47 @@
...
+      public static final double kFRTurnOffset = 0.720806;
+      public static final double kBLTurnOffset = 0.198625;
+      public static final double kBRTurnOffset = 0.425397;
+
+      public static final int kFLTurnMotorId = 9;
+      public static final int kFRTurnMotorId = 10;
+      public static final int kBLTurnMotorId = 11;
+      public static final int kBRTurnMotorId = 12;
+
+      public static final int kFLTurnEncoderChannelA = 0;
```
Comment (snydecor000) [inline src/main/java/frc/robot/Constants.java]: All of these TurnEncoder channels are unused

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -52,31 +55,72 @@ public void robotPeriodic() {
 
   @Override
   public void autonomousInit() {
+    m_swerve.resetKinematics();
   }
```
Comment (snydecor000) [inline src/main/java/frc/robot/Robot.java:62]: Get rid of the resetKinematics(); function definition and calls

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -3,80 +3,119 @@
...
+
+    // m_swerve.drive(0.3, 0, 0, false);
+    // m_swerve.drive(0, 0.1, 0, false);
+    // m_swerve.drive(0, 0, 0.1, false);
+
+    // Intake controls
+    /*
+     * if (mDriverController.getWantsIntakeOpen()) {
+     * // m_intake.open();
+     * } else if (mDriverController.getWantsIntakeClose()) {
```
Comment (GingerJakeDaBoi) [inline src/main/java/frc/robot/Robot.java:105]: mjfjfmjfjjff

**Review by tricktrap — APPROVED** [review]

**Pull request merged**